core/src/main/java/io/grpc/ForwardingDelegateChannelBuilder.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link ManagedChannelBuilder} that delegates all its builder method to another builder by
+ * default.
+ *
+ * <p>A subclass of it typically has the following signature:
+ *
+ * <pre>
+ * public class CustomChannelBuilder
+ *     extends ForwardingDelegateChannelBuilder&lt;CustomChannelBuilder, NettyChannelBuilder&gt;
+ * </pre>
+ *
+ * @param <T> The type of the subclass extending this abstract class.
+ * @param <DelegateT> The type of the delegate.
+ *
+ * @since 1.7.0
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/3363")
+public abstract class ForwardingDelegateChannelBuilder<
```

Comment:
I also thought about `ForwardingChannelBuilder` first, but worried that it may not be as accurate in meaning. I don't mind to use `ForwardingChannelBuilder` though.